terraform-provider-constellation/internal/provider/convert.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package provider
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/edgelesssys/constellation/v2/internal/api/attestationconfigapi"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/idkeydigest"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/measurements"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/variant"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+)
+
+func parseAttestationConfig(tfMeasurements map[string]measurement, tfSnpAttestation sevSnpAttestation, attestationVariant variant.Variant) (config.AttestationCfg, error) {
```

Comment:
We should explain what this function does. It's not clear to me at first glance. I see that it converts values coming from Terraform to Constellation-compatible values by looking at the body. Maybe we could add that as a doc comment?